cli/src/init.rs

```diff
@@ -0,0 +1,238 @@
+use dialoguer::{theme::ColorfulTheme, Input, MultiSelect};
+use include_dir::{include_dir, Dir};
+use std::env;
+use std::error::Error;
+use std::fs::{self, File};
+use std::io::Write;
+use std::path::Path;
+
+pub fn init_project(
+    arg_adapter: &Option<String>,
+    arg_project_name: &Option<String>,
+) -> Result<(), Box<dyn Error>> {
+    let project_name: String = match arg_project_name.as_deref() {
+        None => Input::with_theme(&ColorfulTheme::default())
+            .with_prompt("Project name")
+            .with_initial_text("mopro-example-app".to_string())
+            .interact_text()?,
+        Some(name) => name.to_string(),
+    };
+
+    let adapters = vec!["circom", "halo2"];
+
+    let selection = match arg_adapter.as_deref() {
+        None => MultiSelect::with_theme(&ColorfulTheme::default())
+            .with_prompt("Pick the adapters you want to use (multiple selection with space)")
+            .items(&adapters)
+            .interact()
+            .unwrap(),
+        Some(a) => {
+            let mut selection = vec![];
+            for (i, adapter) in adapters.iter().enumerate() {
+                if a.contains(adapter) {
+                    selection.push(i);
+                }
+            }
+            selection
+        }
+    };
+
+    if selection.is_empty() {
+        println!("\x1b[33mNo adapters selected. Use space to select an adapter.\x1b[0m");
```

Comment:
I'd recommend using something like [this](https://crates.io/crates/console) or [this](https://crates.io/crates/yansi) to handle coloring/formatting terminal text.

Reply:
[3bbec3f](https://github.com/zkmopro/mopro/pull/235/commits/3bbec3fcbbacd7a329e6441a68257eac1a0bf193)